### PR TITLE
Correct Read The Docs builds

### PR DESCRIPTION
**Pull request by portante**
Starting around August 22nd, 2022, the Read The Docs builds began failing, likely due to the obsolescence of the `sphinx-panels` module.

While verifying the build was fixed, we corrected a number of problems found with `Makefile` targets, the formatting of the configuration file, two warnings in the user guide, and dropped the Travis-CI documentation entirely.

### CODE REVIEW

**Comment by portante** [comment]
I have verified the build using a `PBENCH-914` branch enabled on the public repository, see https://readthedocs.org/projects/pbench/builds/18085177/.

**Review by portante — COMMENTED** [review]
Ready for review.

File: docs/Makefile [hunk 1]
```diff
@@ -5,7 +5,7 @@
 all:  pbench-agent.html
```
Comment (webbnh) [inline docs/Makefile:6]: Quick question:  re line 41, is it obvious from context _which_ `requirements.txt` file that is?  (I.e., it's the one in `.../doc/`, and _not_ the one at the base of the Git checkout...which wasn't obvious to _me_...but, perhaps I'm not part of the target demographic....)

(While we're tweaking this file, should we clarify that?)

File: docs/conf.py [hunk 2]
```diff
@@ -94,7 +93,7 @@
 # Add any paths that contain custom static files (such as style sheets) here,
 # relative to this directory. They are copied after the builtin static files,
 # so a file named "default.css" will overwrite the builtin "default.css".
-html_static_path = ["_static"]
+# html_static_path = ["_static"]
```
Comment (webbnh) [inline docs/conf.py:96]: Should there be a blank (comment) line before line 96, to match the ones at, e.g., line 90 and 105?

**Review by webbnh — APPROVED** [review]
Looks good to me.

If you don't feel like addressing the two nits I found, I'll just resolve the conversations.

File: docs/Makefile (hunk 1 above)
Comment (dbutenhof) [reply docs/Makefile:6]: Hmm; should that include a `cd doc`??

**Review by dbutenhof — APPROVED** [review]

File: docs/Makefile (hunk 1 above)
Comment (webbnh) [reply docs/Makefile:6]: Makefiles in general, and ours in specific, tend to assume that the files are referenced relative to the directory where the `Makefile` is.  That is, whoever is running this `make` already did the `cd doc` (or, did a `make -C doc`, which is pretty much the same thing).

Now, whether the output of this "build" should be going into our _source_ directory is another question altogether...but that's basically our established practice....  🥴

Comment (portante) [reply docs/Makefile:6]: I've updated the reference to be `docs/requirements.txt`.

**Review by webbnh — APPROVED** [review]
👍

**Review by ndokos — APPROVED** [review]

**Pull request merged**